<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>424</width>
    <height>415</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>390</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Visual Command</string>
  </property>
  <property name="windowIcon">
   <iconset resource="Images.qrc">
    <normaloff>:/new/prefix1/icons/PNG/onebit_04.png</normaloff>:/new/prefix1/icons/PNG/onebit_04.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="sizeConstraint">
     <enum>QLayout::SetMaximumSize</enum>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Default</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2"/>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>50</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>50</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QFormLayout" name="formLayout">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::ExpandingFieldsGrow</enum>
         </property>
         <item row="0" column="0">
          <widget class="QPushButton" name="pushButtonRun">
           <property name="text">
            <string>Run</string>
           </property>
           <property name="icon">
            <iconset resource="Images.qrc">
             <normaloff>:/new/prefix1/icons/PNG/onebit_27.png</normaloff>:/new/prefix1/icons/PNG/onebit_27.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPlainTextEdit" name="plainTextEditArgumentText">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="plainText">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>424</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionRunMode"/>
   <addaction name="actionSearch"/>
   <addaction name="actionEdit_Mode"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBarEdit">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSave"/>
   <addaction name="actionUpload_GUI"/>
   <addaction name="separator"/>
   <addaction name="actionAdd_Tab"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QDockWidget" name="dockWidgetProperties">
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout_3"/>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidgetControls">
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QListWidget" name="listWidgetArguments"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionRunMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/new/prefix1/icons/PNG/onebit_04.png</normaloff>:/new/prefix1/icons/PNG/onebit_04.png</iconset>
   </property>
   <property name="text">
    <string>RunMode</string>
   </property>
   <property name="toolTip">
    <string>Switches to Run Mode</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionEdit_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/new/prefix1/icons/PNG/onebit_20.png</normaloff>:/new/prefix1/icons/PNG/onebit_20.png</iconset>
   </property>
   <property name="text">
    <string>Edit Mode</string>
   </property>
   <property name="toolTip">
    <string>Switches to Edit Mode</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionSearch">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/new/prefix1/icons/PNG/onebit_02.png</normaloff>:/new/prefix1/icons/PNG/onebit_02.png</iconset>
   </property>
   <property name="text">
    <string>Search</string>
   </property>
   <property name="toolTip">
    <string>Search for a script</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionUpload_GUI">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/new/prefix1/icons/PNG/onebit_30.png</normaloff>:/new/prefix1/icons/PNG/onebit_30.png</iconset>
   </property>
   <property name="text">
    <string>Upload GUI</string>
   </property>
   <property name="toolTip">
    <string>Uploads the current GUI</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="actionAdd_Tab">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/new/prefix1/icons/PNG/onebit_13.png</normaloff>:/new/prefix1/icons/PNG/onebit_13.png</iconset>
   </property>
   <property name="text">
    <string>Add Tab</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/new/prefix1/icons/PNG/onebit_11.png</normaloff>:/new/prefix1/icons/PNG/onebit_11.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="toolTip">
    <string>Save a hardcopy of this script</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="Images.qrc">
     <normaloff>:/new/prefix1/icons/PNG/32px-Crystal_Clear_app_package.png</normaloff>:/new/prefix1/icons/PNG/32px-Crystal_Clear_app_package.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="Images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionSearch</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>LaunchSearch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdit_Mode</sender>
   <signal>triggered()</signal>
   <receiver>actionRunMode</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRunMode</sender>
   <signal>triggered()</signal>
   <receiver>actionEdit_Mode</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdit_Mode</sender>
   <signal>toggled(bool)</signal>
   <receiver>toolBarEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>251</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionUpload_GUI</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>LaunchUpload()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_Tab</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>EditAddTab()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>210</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>EditRemoveTab(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>65</x>
     <y>56</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRunMode</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>SwitchMode(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listWidgetArguments</sender>
   <signal>itemDoubleClicked(QListWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>EditAddWidget(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>FileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>FileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonRun</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>RunCommand()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>363</y>
    </hint>
    <hint type="destinationlabel">
     <x>46</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSettings</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>EditSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>LaunchAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>LaunchSearch()</slot>
  <slot>RunCommand()</slot>
  <slot>LaunchUpload()</slot>
  <slot>EditAddTab()</slot>
  <slot>EditRemoveTab(int)</slot>
  <slot>EditAddWidget(QListWidgetItem*)</slot>
  <slot>SwitchMode(bool)</slot>
  <slot>FileOpen()</slot>
  <slot>FileSave()</slot>
  <slot>EditSettings()</slot>
  <slot>LaunchAbout()</slot>
 </slots>
</ui>
